src/lava/lib/optimization/solvers/generic/scif/models.py

```diff
@@ -242,29 +385,36 @@ def _gen_sig_spks(self, spk_hist_buffer):
         # Compute the local cost
         s_sig[sig_spk_idx] = self.cost_diagonal[sig_spk_idx] + \
             self.a_in_data[sig_spk_idx]
```

Comment:
Does this line imply that all active neurons send out their local cost within the WTA in each time step? If so, worth sparsifying it. It is sufficient if neurons send their local cost within the WTA only when their state has changed from 0 <-> 1 (delta coding).